apps/condo/domains/ticket/components/CallRecordModal.tsx

```diff
@@ -0,0 +1,226 @@
+
+    const handleAttachTicket = useCallback((ticketIds) => {
+        setAttachedTickets(ticketIds)
+    }, [])
+
+    const resetAttachedTickets = useCallback(() => {
+        setSelectedCallRecordFragment(null)
+        setAttachedTickets([])
+    }, [setSelectedCallRecordFragment])
+
+    const handleSave = useCallback(() => {
+        const removedTickets = difference(initialAttachedTickets, attachedTickets)
+        const newAttachedTickets = difference(attachedTickets, initialAttachedTickets)
+
+        removedTickets.forEach((ticketId, index) => {
+            const callRecordFragment = initialCallRecordFragments
+                .find(fragment => get(fragment, 'ticket.id') === ticketId)
+            if (!callRecordFragment) return
+
+            const emptyCallRecordFragment = initialCallRecordFragments
+                .find(fragment =>
+                    get(fragment, 'callRecord.id') === callRecordId &&
+                    isEmpty(get(fragment, 'ticket'))
+                )
+
+            if (
+                index === 0 &&
+                removedTickets.length === initialAttachedTickets.length &&
+                isEmpty(newAttachedTickets) &&
+                !emptyCallRecordFragment
+            ) {
+                return updateCallRecordFragment({ ticket: { disconnect: { id: ticketId } } }, callRecordFragment)
+            }
+
+            return softDeleteCallRecordFragment(callRecordFragment)
+        })
+
+        newAttachedTickets.forEach((ticketId, index) => {
+            if (index === 0 && isEmpty(initialAttachedTickets)) {
+                const emptyCallRecordFragment = initialCallRecordFragments
+                    .find(fragment => get(fragment, 'callRecord.id') === callRecordId)
+
+                if (emptyCallRecordFragment) {
+                    return updateCallRecordFragment({ ticket: { connect: { id: ticketId } } }, emptyCallRecordFragment)
+                }
+            }
+
+            return createCallRecordFragment({
+                callRecord: { connect: { id: callRecordId } },
+                ticket: { connect: { id: ticketId } },
+                startedAt: get(selectedCallRecordFragment, 'startedAt'),
+            })
+        })
+
+        resetAttachedTickets()
+    }, [attachedTickets, callRecordId, createCallRecordFragment, initialAttachedTickets,
+        initialCallRecordFragments, resetAttachedTickets, selectedCallRecordFragment, softDeleteCallRecordFragment, updateCallRecordFragment])
+
+    const closeModal = useCallback(() => {
+        resetAttachedTickets()
+    }, [resetAttachedTickets])
+
+    const { downloadFile } = useDownloadFileFromServer()
+
+    const url = get(selectedCallRecordFragment, 'callRecord.file.publicUrl')
+    const name = get(selectedCallRecordFragment, 'callRecord.file.originalFilename')
+    const handleDownloadFile = () => {
+        downloadFile({ url, name })
+    }
+
+    if (!selectedCallRecordFragment || loading) {
+        return <></>
```

Comment:
If this leftover?

Use null, if you want explicitly say to render nothing (null)